src/validator.js

```diff
@@ -0,0 +1,64 @@
+import * as fs from 'fs';
+import Xpi from  'xpi';
+
+import promisify from 'es6-promisify';
+import chalk from 'chalk';
+
+export var lstat = promisify(fs.lstat);
+
+
+export default class Validator {
+
+  constructor(config) {
+    this.config = config;
+    this.packagePath = config._[0];
+    this.xpi;
+    this.chalk = new chalk.constructor(
+      {enabled: !this.config.boring});
+  }
+
+  handleError(err) {
+    if (this.config.stack === true) {
+      console.error(err.stack);
+    } else {
+      console.error(this.chalk.red(err.message || err));
+    }
+  }
+
+  checkFileExists(filepath, _lstat=lstat) {
+    var invalidMsg = new Error(
+      'Path given is not a file or does not exist: "' + filepath + '"');
```

Comment:
We should use template strings here instead, taking care of ES6 niceness. I think we did this elsewhere and I forgot to mention it, but:

```js
var invalidMessage = new Error(
  `Path given is not a file or does not exist: "${filepath}"`);
```
should work. Also, `invalidMessage` is a nicer variable name, we have space :smile: